src/lib.rs

```diff
@@ -282,6 +282,60 @@ pub fn fbm_simplex_2d_seeded(
     sum
 }
 
+/// The return value of the domain warping function.
+#[derive(Copy, Clone)]
+pub struct WarpResult {
+    /// The final noise value
+    pub noise_value: f32,
+    /// The history of warped coordinates. Can be useful for mixing colors.
+    pub positions: [Vec2; 4],
```

Comment:
I think the behavior for this field is a bit confusing when `warp_iterations > 4`, it seems to be keeping the first three and the last position.

I wonder if there is a better way of exposing this. It does look really cool when used for coloring :)

Reply:
Great idea, as a user it feels better to access the warp position history like this `position[0]`, `position[1]`.